File: front_end/vegetable/src/service/cart/CartService.js
import axios from "axios";
export const getAllCartByUserName = async (userName) =>{
    try{
        const response = await axios.get(`http://localhost:8080/api/v1/auth/cart?userName=${userName}`);
        return response.data;
    }catch(e){
        console.log(e);
    }
}
export const deleteCartByCartId = async (productId,userId) =>{
    try{
        await axios.delete(`http://localhost:8080/api/v1/auth/delete?productId=${productId}&userId=${userId}`);
    }catch(e){
        console.log(e);
    }
}
export const createCart = async (quantityProductOrder,productId,userName) =>{
    try{
        await axios.post(`http://localhost:8080/api/v1/auth/create?quantityProductOrder=${quantityProductOrder}&productId=${productId}&userName=${userName}`);
    }catch(e){
        console.log(e);
    }
}

File: back_end/not/src/main/java/com/example/not/service/cart/CartService.java
package com.example.not.service.cart;
import com.example.not.IProjection.cart.ICartProjection;
import com.example.not.model.user.Users;
import com.example.not.repository.cart.ICartRepository;
import com.example.not.repository.users.IUserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import java.util.List;

@Service
public class CartService implements ICartService{
    @Autowired
    private ICartRepository iCartRepository;
    @Autowired
    private IUserRepository iUserRepository;

    @Override
    public List<ICartProjection> findAllCartByUserName(String userName) {
        return iCartRepository.findAllCartByUserName(userName);
    }

    @Override
    public void createCart(int quantityProductOrder,int productId,String userName) {
        iCartRepository.createCart(quantityProductOrder,productId,iUserRepository.findByUserName(userName).getUserId());
    }

    @Override
    public void deleteCartByCartId(int productId,int userId) {
        iCartRepository.deleteCartByIdCart(productId,userId);
    }

}
